tests/integration/cartography/intel/github/test_repos.py

```diff
@@ -8,423 +10,398 @@
+        "homepage",
+        "defaultbranch",
+        "defaultbranchid",
+        "private",
+        "disabled",
+        "archived",
+        "locked",
+        "giturl",
+        "url",
+        "sshurl",
+        "createdat",
+        "updatedat",
+    ]
+    repo_properties = check_nodes(neo4j_session, "GitHubRepository", core_properties)
+    assert repo_properties is not None
+    assert len(repo_properties) == 5
+
+    # Verify no essential properties are None
+    for repo_data in repo_properties:
+        essential_props = repo_data[:17]  # All properties except optional ones
+        assert all(
+            prop is not None for prop in essential_props[:1]
+        ), "ID should not be None"
+        assert all(
+            prop is not None for prop in essential_props[1:7]
+        ), "Core text properties should not be None"
+        assert all(
+            isinstance(prop, bool) for prop in essential_props[8:12]
+        ), "Boolean flags should be boolean"
+        assert all(
+            prop is not None for prop in essential_props[12:17]
+        ), "URL and date properties should not be None"
+
+    # Test specific property values for sample repository
+    sample_repo = next(
+        props
+        for props in repo_properties
+        if props[0] == "https://github.com/simpsoncorp/sample_repo"
     )
-    nodes = neo4j_session.run(
-        "MATCH(owner:GitHubOrganization) RETURN owner.id",
+    (
+        repo_id,
+        name,
+        fullname,
+        description,
+        primarylang,
+        homepage,
+        defaultbranch,
+        defaultbranchid,
+        private,
+        disabled,
+        archived,
+        locked,
+        giturl,
+        url,
+        sshurl,
+        createdat,
+        updatedat,
+    ) = sample_repo
+
+    # Validate all properties have expected values
+    assert repo_id == "https://github.com/simpsoncorp/sample_repo"
+    assert name == "sample_repo"
+    assert fullname == "simpsoncorp/sample_repo"
+    assert description == "My description"
+    assert primarylang == "Python"
+    assert homepage == ""
+    assert defaultbranch == "master"
+    assert (
+        defaultbranchid == "https://github.com/simpsoncorp/sample_repo:branch_ref_id=="
     )
-    actual_nodes = {n["owner.id"] for n in nodes}
-    expected_nodes = {
-        "https://github.com/simpsoncorp",
-    }
-    assert actual_nodes == expected_nodes
-
-
-def test_transform_and_load_repository_languages(neo4j_session):
+    assert private is True
+    assert disabled is False
+    assert archived is False
+    assert locked is True
+    # assert giturl == "git://github.com:simpsoncorp:sample_repo.git"  buggy url conversion
+    assert url == "https://github.com/simpsoncorp/sample_repo"
+    assert sshurl == "git@github.com:simpsoncorp/sample_repo.git"
+    assert createdat == "2011-02-15T18:40:15Z"
+    assert updatedat == "2020-01-02T20:10:09Z"
+
+
+def _test_owner_nodes_and_relationships(neo4j_session):
```

Comment:
As I mentioned earlier, let's not use these extra `_test()` functions because to trace a problem and read the test, we need to find what function called this function, and that's harder to do than just simply scrolling up and down in the same function. As currently written, I don't know what things happened to lead up to the set of assertions that are here.